community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/KernelAPIReadTestBase.java

```diff
@@ -44,6 +48,27 @@
 @SuppressWarnings( "WeakerAccess" )
 public abstract class KernelAPIReadTestBase<ReadSupport extends KernelAPIReadTestSupport>
 {
+    public class ConditionalTeardown implements TestRule
```

Comment:
How about calling this class something like `CursorsClosedPostCondition`? Since that is what it does.